lib/AppInfo/Application.php

```diff
@@ -55,5 +58,27 @@ public function __construct() {
 				$Container->query('ImageDimensionMapper')
 			);
 		});
+
+		$container->registerService('RetrieveMetadata', function(IContainer $Container) {
+			return new RetrieveMetadata(
+				$Container->query('ServerContainer')->getDb()
+			);
+		});
+
+		$container->registerService('MetadataController', function(IContainer $container){
+			return new MetadataController(
+				$container->query('AppName'),
+				$container->query('Request'),
+				$container->query('RetrieveMetadata')
+			);
+		});
+
+		$container->registerService('MetadataApiController', function(IContainer $container){
+			return new MetadataApiController(
+				$container->query('AppName'),
+				$container->query('Request'),
+				$container->query('RetrieveMetadata')
+			);
+		});
```

Comment:
For the 2nd part, I was wondering if you had checked the database when doing the tests to see if the metadata was properly stored since it doesn't work when I run it here.